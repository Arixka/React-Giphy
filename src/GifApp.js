import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifApp = ({defaultCategories = []}) => {

  const [categories, setCategories] = useState(defaultCategories)

  //console.log(categories)

  return <>
    <h2>Giphy App</h2>
    <AddCategory setCategories={setCategories} />
    <hr />
    {/* <button onClick={handleAdd}>Agregar</button> */}
    <ol>
      {categories.map(category =>
        <GifGrid 
        category={category.toUpperCase()} 
        key={category} />
        )}
    </ol>


  </>
}
