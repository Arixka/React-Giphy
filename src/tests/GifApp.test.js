import React from 'react'
import { shallow } from 'enzyme'
import { GifApp } from '../GifApp'

describe('pruebas con gifApp', () => {
  const wrapper = shallow(<GifApp />)

  test('debe renderizar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de mostrar una lista de categorias', () => {
    const categories = ['cats', 'dogs']
    const wrapper = shallow(<GifApp defaultCategories={categories}/>)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })

})