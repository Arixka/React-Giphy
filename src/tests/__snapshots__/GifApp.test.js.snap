// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con gifApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Giphy App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="CATS"
      key="cats"
    />
    <GifGrid
      category="DOGS"
      key="dogs"
    />
  </ol>
</Fragment>
`;

exports[`pruebas con gifApp debe renderizar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Giphy App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
